### PR TITLE
Bugfix FXIOS-9667 Reduce latency loading favicons

### PR DESCRIPTION
**Pull request by ih-codes**
## :scroll: Tickets
[Jira ticket 1](https://mozilla-hub.atlassian.net/browse/FXIOS-9667)
[Github issue 1](https://github.com/mozilla-mobile/firefox-ios/issues/21273)

[Jira ticket 2](https://mozilla-hub.atlassian.net/browse/FXIOS-9690)
[Github issue 2](https://github.com/mozilla-mobile/firefox-ios/issues/21307)

## :bulb: Description

Two tickets linked, because the favicon rework solves the bug in ticket 1.

**Context for the favicon rework:**
On the very first launch of the app, the home page top sites (in the pinned area) show the 5 DefaultSuggestedSites, the Google tile, and any sponsored tiles by default.

While the app bundled favicons with the intention of reducing load times of fetching and showing favicons for each of these top sites, they were not being used as expected. (bug) This meant the first launch was slow to show the favicons in the home page tiles. Moreover, several of the bundled favicons were very outdated anyway.

The consequence is that the app would fire off a network request for each website to scrape the faviconURL, which would then be used to make an image request.

The app caches the URLs and images to reduce load times on future launches.

**Main changes:**
- Removed all the bundled favicon assets, many of which were outdated
- Removed code and tests for loading these bundled assets (no longer necessary)
- Hard coded DefaultSuggestedSites faviconURLs to reduce latency looking up the top sites favicon URLs on first app launch

The bugfix ensures that the hard coded favicon URL is used for the top suggested sites. This means requests do not have to be made to each of the DefaultSuggestedSites to scrape the favicon URLs from the metadata (these requests took around ~400ms on my internet).

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by ih-codes** [comment]
@OrlaM I just wanted to make a quick draft PR since you'll be away on PTO next week.

I think most of the changes are here, but I'm still doing some testing. 

I also am talking to @mattreaganmozilla about the tab metadata providing the wrong faviconURL in some cases (e.g. user goes to `gmail.com`, which redirects to `accounts.google.com/v3/signin/identifier?...`, and a bad faviconURL is returned from the JavaScript evaluation). Whereas the `fetchFaviconURL` method comes up with a correct guess to the default favicon.ico link. Still investigating, but might be a race condition which wins out. Seems like this didn't occur before because of the bundled assets.

File: firefox-ios/Storage/DefaultSuggestedSites.swift [hunk 1]
```diff
@@ -37,83 +37,64 @@ open class DefaultSuggestedSites {
...
-                imageUrl: "asset://suggestedsites_twitter",
-                faviconUrl: "asset://defaultFavicon",
+                url: "https://x.com/", // TODO Update Twitter to X
+                faviconUrl: "https://abs.twimg.com/responsive-web/client-web/icon-ios.77d25eba.png",
                 trackingId: 628,
                 title: .DefaultSuggestedTwitter
             )
         ],
-        "zh_CN": [
+        "zh_CN": [ // FIXME Do we still want this as a special case localization? Android doesn't compile this vers. anymore
```
[the remark this answers is not in the document]
Comment (OrlaM) [reply firefox-ios/Storage/DefaultSuggestedSites.swift:70]: I haven't had a chance to chase down an answer for this. I'd say ignore it as best you can for now even if it means leaving it in a slightly broken state. I'll try track down an answer for this when I get back. There are a few other areas we can clean up if we no longer care about this version.

**Review by OrlaM — APPROVED** [review]
LGTM

**Comment by ih-codes** [comment]
> > I also am talking to @mattreaganmozilla about the tab metadata providing the wrong faviconURL in some cases (e.g. user goes to `gmail.com`, which redirects to `accounts.google.com/v3/signin/identifier?...`, and a bad faviconURL is returned from the JavaScript evaluation). Whereas the `fetchFaviconURL` method comes up with a correct guess to the default favicon.ico link. Still investigating, but might be a race condition which wins out. Seems like this didn't occur before because of the bundled assets.
> 
> [Matt created a ticket for this](https://mozilla-hub.atlassian.net/browse/FXIOS-9745).

I'm hesitant to merge this until this issue has been addressed, because it could make the problem look worse.

File: firefox-ios/Storage/DefaultSuggestedSites.swift [hunk 2]
```diff
@@ -37,83 +37,64 @@ open class DefaultSuggestedSites {
...
+                faviconUrl: "https://m.youtube.com/static/apple-touch-icon-180x180-precomposed.png",
                 trackingId: 631,
                 title: .DefaultSuggestedYouTube
             ),
             SuggestedSiteData(
                 url: "https://www.amazon.com/",
-                bgColor: "0x000000",
-                imageUrl: "asset://suggestedsites_amazon",
-                faviconUrl: "asset://defaultFavicon",
+                // FIXME Need the correct favicon URL... grabbed this one from the contile services link
```
Comment (ih-codes) [inline firefox-ios/Storage/DefaultSuggestedSites.swift:52]: Amazon famously [doesn't have an Apple Touch Icon](https://www.reddit.com/r/MacOS/comments/qhl25n/why_is_amazon_like_this/)...

Is it safe for us to use the contile link as the favicon URL? I got it from https://contile.services.mozilla.com/v1/tiles

Amazon.com only hosts a low quality .ico and an Open Graph image that's the full "Amazon" text (not ideal for favicon sizes).

**Comment by mattreaganmozilla** [comment]
>> I also am talking to @mattreaganmozilla about the tab metadata providing the wrong faviconURL in some cases (e.g. user goes to gmail.com, which redirects to accounts.google.com/v3/signin/identifier?..., and a bad faviconURL is returned from the JavaScript evaluation). Whereas the fetchFaviconURL method comes up with a correct guess to the default favicon.ico link. Still investigating, but might be a race condition which wins out. Seems like this didn't occur before because of the bundled assets.
>>    [Matt created a ticket for this](https://mozilla-hub.atlassian.net/browse/FXIOS-9745).

> I'm hesitant to merge this until this issue has been addressed, because it could make the problem look worse.

Let me know if you have a minute to sync on this. I still need to revisit that ticket, it's on my radar but I just haven't had time yet to get back to it but I can try to prioritize it if it's blocking this work. It's possible that we might just need to apply some kind of bandaid temporarily (like directly examining the URL the JS spits out to see if the .ico is in a subdirectory incorrectly).

**Comment by ih-codes** [comment]
Ok, the aforementioned ticket that was holding this up [has been "resolved"](https://mozilla-hub.atlassian.net/browse/FXIOS-9745). Thanks @mattreaganmozilla !

@OrlaM You reviewed this already before mozweek, but just FYI there are two comments awaiting your input (no rush though, I know you're still catching up on a bunch of stuff).

- https://github.com/mozilla-mobile/firefox-ios/pull/21405/files#r1727507576
- https://github.com/mozilla-mobile/firefox-ios/pull/21405/files#r1702088030

**Pull request merged**